File: 0926 Flip String to Monotone Increasing.py
class Solution:
    def minFlipsMonoIncr(self, S: str) -> int:
        f0, f1 = 0, 0
        for ch in S:
            if ch == '0':
                f1 += 1
            else:
                f1 = min(f0, f1)
                f0 += 1
        return min(f0, f1)

File: 0926 Flip String to Monotone Increasing.java
class Solution {
    public int minFlipsMonoIncr(String S) {
        int f0 = 0, f1 = 0;
        for (char c: S.toCharArray()) {
            if (c == '0') {
                ++f1;
            } else {
                f1 = Math.min(f0, f1);
                ++f0;
            }
        }
        return Math.min(f0, f1);
    }
}